torch_xla/csrc/tensor.cpp

```diff
@@ -242,15 +242,15 @@ void XLATensor::SetShardingSpec(const ShardingSpec& sharding) {
         << ", must be cleared before applying a new one, "
         << sharding.sharding.DebugString();
   }
-  dynamic_cast<XlaNode*>(GetIrValue().node.get())
-      ->SetSharding(sharding_spec()->sharding);
+  CreateShardedIrValue(sharding_spec());
 }
+
 void XLATensor::ClearShardingSpec() {
   data()->sharding = nullptr;
   torch::lazy::Value ir_value = CurrentIrValue();
   if (ir_value) {
     // This should be a no-op if there is no sharding.
-    dynamic_cast<XlaNode*>(ir_value.node.get())->ClearSharding();
+    CreateUnshardedIrValue();
```

Comment:
ditto

Reply:
Done

Reply:
it's little verbose, but I actually preferred `dynamic_cast<XlaNode*>(ir_value.node.get())->XYZ` to show that the tensor method `ClearShardingSpec` calls the underlying node's method.